File: app/src/main/java/in/mvpstarter/sample/ui/detail/DetailMvpView.java
package in.mvpstarter.sample.ui.detail;

import in.mvpstarter.sample.data.model.Pokemon;
import in.mvpstarter.sample.data.model.Statistic;
import in.mvpstarter.sample.ui.base.MvpView;

public interface DetailMvpView extends MvpView {

    void showPokemon(Pokemon pokemon);

    void showStat(Statistic statistic);

    void showProgress(boolean show);

    void showError(Throwable error);

}

File: app/src/main/java/in/mvpstarter/sample/ui/detail/DetailMvpView.kt
package `in`.mvpstarter.sample.ui.detail

import `in`.mvpstarter.sample.data.model.Pokemon
import `in`.mvpstarter.sample.data.model.Statistic
import `in`.mvpstarter.sample.ui.base.MvpView

interface DetailMvpView : MvpView {

    fun showPokemon(pokemon: Pokemon)

    fun showStat(statistic: Statistic)

    fun showProgress(show: Boolean)

    fun showError(error: Throwable)

}